R/zTest.R

```diff
@@ -0,0 +1,706 @@
+
+      estimate <- mean(x-y)
+      names(estimate) <- "mean of the differences"
+      zStat <- tryOrFailWithNA(sqrt(n1)*estimate/sigma)
+    } else {
+      testType <- "twoSampleZ"
+
+      nEff <- (1/n1+1/n2)^(-1)
+      estimate <- c(mean(x), mean(y))
+      names(estimate) <- c("mean of x", "mean of y")
+      zStat <- tryOrFailWithNA(sqrt(nEff)*(estimate[1]-estimate[2])/sigma)
+    }
+  }
+
+  result[["testType"]] <- testType
+
+  if (is.na(zStat))
+    stop("Could not compute the z-statistic")
+
+  if (pilot) {
+    nPlan <- if (is.null(n2)) n1 else c(n1, n2)
+
+    designObj <- designPilotSafeZ("alpha"=alpha, "nPlan"=nPlan, "alternative"=alternative, "h0"=h0,
+                                  "sigma"=sigma, "paired"=paired, "tol"=tol)
+    designObj[["pilot"]] <- TRUE
+  }
+
+  if (designObj[["testType"]] != testType)
+    warning('The test type of designObj is "', designObj[["testType"]],
+            '", whereas the data correspond to a testType "', testType, '"')
+
+  if (alternative != designObj[["alternative"]])
+    warning('The test is designed for a test with direction "', designObj[["alternative"]],
+            '", whereas the requested analysis has direction "', alternative, '"')
+
+  sValue <- safeZTestStat("z"=zStat, "parameter"=designObj[["parameter"]], "n1"=n1, "n2"=n2,
+                          "alternative"=alternative, "paired"=paired)
+
+  if (is.null(y))
+    dataName <- as.character(sys.call())[2]
+  else
+    dataName <- paste(as.character(sys.call())[2], "and", as.character(sys.call())[3])
+
+  # TODO(Alexander): Add sCIs
+
+  names(zStat) <- "z"
+
+  result[["statistic"]] <- zStat
+  result[["estimate"]] <- estimate
+  result[["dataName"]] <- dataName
+  result[["designObj"]] <- designObj
+
+  if (is.null(n2)) {
+    result[["n"]] <- n1
+    names(result[["n"]]) <- "n1"
+  } else {
+    result[["n"]] <- c(n1, n2)
+    names(result[["n"]]) <- c("n1", "n2")
+  }
+  result[["sValue"]] <- sValue
+
+  return(result)
+}
+
+#' Alias for safeZTest
+#'
+#' @rdname safeZTest
+#'
+#' @export
+safe.z.test <- function(x, y=NULL, designObj=NULL, alternative=c("two.sided", "less", "greater"),
+                        h0=0, sigma=1, paired=FALSE, confLevel=1-alpha, pilot=FALSE, alpha=0.05, ...) {
+  result <- safeZTest("x"=x, "y"=y, "designObj"=designObj, "alternative"=alternative,
+                      "h0"=h0, "paired"=paired, "sigma"=sigma, "confLevel"=confLevel, "pilot"=pilot, "alpha"=alpha, ...)
+
+  if (is.null(y))
+    dataName <- as.character(sys.call())[2]
+  else
+    dataName <- paste(as.character(sys.call())[2], "and", as.character(sys.call())[3])
+
+  result[["dataName"]] <- dataName
+  return(result)
+}
+
+#' Design a Frequentist Z-Test
+#'
+#' Computes the number of samples necessary to reach a tolerable type I and type II error for the
+#' frequentist z-test.
+#'
+#' @inheritParams designSafeZ
+#'
+#' @return returns a freqZDesign object.
+#' @export
+#'
+#' @examples
+#' designFreqT(0.5)
```

Comment:
Hier nog even juist voorbeeld toevoegen? 
```suggestion
#' testResult <- designFreqZ(meanDiffMin = 0.5, highN = 100)
#' testResult$nPlan
#' testResult2 <- designFreqZ(meanDiffMin = 0.2, lowN = 32, highN = 200)
#' testResult2$nPlan
```